packages/kinvey-datastore/src/persisters/key-value-persister.js

```diff
@@ -0,0 +1,67 @@
+const _cache = {};
+
+export class KeyValuePersister {
+  /** @type {Boolean} */
+  _cacheEnabled;
+  /** @type {Number} */
+  _ttl;
+
+  // TODO: implement TTL, make _cache a constructor argument?
```

Comment:
This TTL is not the TTL you are referring to. This has to do with reading and "caching" the read information, in order to avoid reading it every time from persistance. You may remember I told you about it when I added it.

Until now, for each operation, the inmemory adapters would read everything, change it, and save it back. This is to avoid reading it. We can play around with the saving too, but as it is out of scope, I've left the TTL out. But this is a change. It should not be felt, but it is there. We can remove it, I guess, but as I did explain what and why at the time, I thought we decided to leave it in.

Reply:
The change will be useful. I like the idea of having an in-memory cache in the persister. I misunderstood TTL before, like you pointed out.

Given your explanation, either we implement the TTL now, or we create a JIRA ticket to address it later. In any case, this "TODO" should be replaced by a ticket, or a solution.

Reply:
I logged ticket [MLIBZ-2236](https://kinvey.atlassian.net/browse/MLIBZ-2236) for this. It's not for implementation, but for a decision, since I think TTL/caching might not be the best way to approach this.